### ISSUE TITLE
Ship test adapter 🙏

### ISSUE THREAD

**Issue by KeesCBakker** [hubotio/hubot#1743]
Great job on the rewrite to `.mjs`, what a task! As I'm upgrading I found my dependencies on [hubot-mock-adapter](https://github.com/hubotio/hubot-mock-adapter) breaking.

When clicking through the Hubot code I've found that Hubot has it's own fixture for it: https://github.com/hubotio/hubot/blob/main/test/fixtures/MockAdapter.mjs, which looks very very very similar to what that package was doing: https://github.com/hubotio/hubot/blob/main/test/fixtures/MockAdapter.mjs

My proposal: shall we ship this test adapter in the NPM package?

**Comment by joeyguerra** [comment]
The intent is to reduce external dependencies. Can you post the problematic code?

**Comment by KeesCBakker** [comment]
It is a bit hard, because it is a chain:

- I'm trying to upgrade the peer dependency on Hubot on my [hubot-command-mapper](https://github.com/KeesCBakker/hubot-command-mapper)
- This one has tests with [hubot-mock-adapter](https://github.com/hubotio/hubot-mock-adapter)
- Which runs on 9, but when I upgrade it starts to sob about peer dependencies on 9
- When I forked and upgraded hubot-mock-adapter, I got into more trouble.
- When I looked into it, I saw the the code was already in the hubot package, hence the thought of shipping it.

But this might all be fixed now that we ship it with our package. So now I'm going back to see if I can drop the dependency on the hubot-mock-adapter.

I have a blog post that uses the hubot-mock-adapter. I'll be checking what I'll need to change over there as well: https://keestalkstech.com/2023/08/hubot-testing-revisited/
